src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonToPackageReferenceMigrator.cs

```diff
@@ -23,34 +22,28 @@ internal class VsProjectJsonToPackageReferenceMigrator : IVsProjectJsonToPackage
         public VsProjectJsonToPackageReferenceMigrator(Lazy<IVsSolutionManager> solutionManager)
         {
             _solutionManager = solutionManager;
-            if(solutionManager == null)
+            if (solutionManager == null)
             {
                 throw new ArgumentNullException(nameof(solutionManager));
             }
         }
-        public IVsProjectJsonToPackageReferenceMigrateResult MigrateProjectJsonToPackageReference(string projectUniqueName)
+        public Task<object> MigrateProjectJsonToPackageReferenceAsync(string projectUniqueName)
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
-            if(string.IsNullOrEmpty(projectUniqueName))
+            if (string.IsNullOrEmpty(projectUniqueName))
             {
                 throw new ArgumentNullException(nameof(projectUniqueName));
             }
 
-            if(!File.Exists(projectUniqueName))
+            if (!File.Exists(projectUniqueName))
             {
                 throw new FileNotFoundException(nameof(projectUniqueName));
             }
 
-            return  NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => 
-            {
-                await TaskScheduler.Default;
-                return await MigrateProjectToPackageRefAsync(projectUniqueName);
-            });
-            
+            return MigrateProjectToPackageRefAsync(projectUniqueName);
         }
 
-        private async Task<IVsProjectJsonToPackageReferenceMigrateResult> MigrateProjectToPackageRefAsync(string projectUniqueName)
-        {            
+        private async Task<object> MigrateProjectToPackageRefAsync(string projectUniqueName)
+        {
             var project = _solutionManager.Value.GetDTEProject(projectUniqueName);
```

Comment:
It's not the same. If you switch first, then the JTF.RUn within GetDTEProject doesn't have to switch and thereby doesn't block a background thread during the switch. So it's at *least* more efficient. Can often avoid long UI delays that happen due to threadpool starvation. And in some very rare cases it can even avoid a deadlock.
This isn't a "must fix" in my book, but it's cheap to fix, and the benefits are real. So I recommend you get in the habit of switching to the UI thread before calling any synchronous code like this that will need the UI thread.